scripts/test.sh

```diff
@@ -97,33 +99,31 @@ for i in "${ignored_packages[@]}"; do
   serial_packages=(${serial_packages[@]//*$i*})
 done
 
-if [ "${1:-""}" = "" ]; then
+if [[ -z "${restrict_testing_to}" ]]; then
   for dir in "${packages[@]}"; do
     pushd "$dir"
       ginkgo -p --race -randomizeAllSpecs -randomizeSuites \
-        -ldflags="-extldflags=-Wl,--allow-multiple-definition" \
-        ${@:2}
+        -ldflags="-extldflags=-Wl,--allow-multiple-definition"
     popd
   done
   for dir in "${serial_packages[@]}"; do
     pushd "$dir"
       ginkgo --nodes "${SERIAL_NODES}" --race -randomizeAllSpecs -randomizeSuites -failFast \
-        -ldflags="-extldflags=-Wl,--allow-multiple-definition" \
-        ${@:2}
+        -ldflags="-extldflags=-Wl,--allow-multiple-definition"
     popd
   done
 else
-  dir="${@: -1}"
+  dir="${restrict_testing_to}"
   dir="${dir#./}"
   for package in "${serial_packages[@]}"; do
     if [[ "${dir##$package}" != "${dir}" ]]; then
-      ginkgo --race -randomizeAllSpecs -randomizeSuites -failFast \
-        -ldflags="-extldflags=-Wl,--allow-multiple-definition" \
-        "${@}"
+      pushd "${dir}"
+      ginkgo --nodes "${SERIAL_NODES}" --race -randomizeAllSpecs -randomizeSuites -failFast \
+        -ldflags="-extldflags=-Wl,--allow-multiple-definition"
       exit $?
     fi
   done
+  pushd "${dir}"
   ginkgo -p --race -randomizeAllSpecs -randomizeSuites -failFast \
-    -ldflags="-extldflags=-Wl,--allow-multiple-definition" \
-    "${@}"
+    -ldflags="-extldflags=-Wl,--allow-multiple-definition"
```

Comment:
[?] Shouldn't this be a loop over the non-serial packages and then running their tests?

Reply:
the loop in the case above is to look at all serial packages and see if the specified test is one of them. if it is, it tests just that package and exits.

if we get here, we know it's a non-serial package, we just run ginkgo against the requested package.